include/bout/region.hxx

```diff
@@ -125,14 +125,66 @@ public:
     ++ind;
     return original;
   }
-  bool operator==(const SpecificInd &x) const { return ind == x.ind; }
-  bool operator!=(const SpecificInd &x) const { return ind != x.ind; }
-  bool operator>(const SpecificInd &x) const { return ind > x.ind; }
-  bool operator>=(const SpecificInd &x) const { return ind >= x.ind; }
-  bool operator<(const SpecificInd &x) const { return ind < x.ind; }
-  bool operator<=(const SpecificInd &x) const { return ind <= x.ind; }
+
+  /// Pre-decrement operator
+  SpecificInd &operator--() {
+    --ind;
+    return *this;
+  }
+
+  /// Post-decrement operator
+  SpecificInd operator--(int) {
+    SpecificInd original(*this);
+    --ind;
+    return original;
+  }
+
+  /// In-place addition
+  SpecificInd &operator+=(SpecificInd n) {
+    ind += n.ind;
+    return *this;
+  }
+
+  /// In-place subtraction
+  SpecificInd &operator-=(SpecificInd n) {
+    ind -= n.ind;
+    return *this;
+  }
+
+  /// Modulus operator
+  SpecificInd operator%(int n) {
+    SpecificInd new_ind(ind % n);
+    return new_ind;
+  }
 };
-// Make identical subclasses with different names
+
+/// Relational operators
+inline bool operator==(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return lhs.ind == rhs.ind;
+}
+inline bool operator!=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator==(lhs, rhs);
+}
+inline bool operator<(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return lhs.ind < rhs.ind;
+}
+inline bool operator>(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return operator<(rhs, lhs);
+}
+inline bool operator>=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator<(lhs, rhs);
+}
+inline bool operator<=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator>(lhs, rhs);
+}
+
+/// Arithmetic operators with integers
+inline SpecificInd operator+(SpecificInd lhs, const SpecificInd &rhs) { return lhs += rhs; }
```

Comment:
Won't this cause problems in uses like the below?

~~~~
Ind3D lhs(4);
Ind3D rhs(5);
auto newLhs = lhs+rhs;
output<<newLhs.ind<<" and "<<lhs.ind<<" should be 9 and 4, but may be 9 and 9 now?\n";
~~~~

Reply:
It takes `lhs` as a copy, and returns the copy. There's a [test][1] explicitly for this :)

[1]: https://github.com/boutproject/BOUT-dev/pull/875/files/896f7d8fc26d48884902d25df444154c09e205de#diff-749772ea70cb4dc76e059e4671c05235R1303

Reply:
Ah yes, didn't spot the absence of &